app/src/main/java/com/doubtless/doubtless/screens/dashboard/usecases/DeleteAccountUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.doubtless.doubtless.screens.dashboard.usecases
+
+import androidx.annotation.WorkerThread
+import com.doubtless.doubtless.constants.FirestoreCollection
+import com.doubtless.doubtless.screens.auth.usecases.UserManager
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class DeleteAccountUseCase(
+    private val firestore: FirebaseFirestore,
+) {
+
+    sealed class Result {
+        class Success(val message: String) : Result()
+        class Error(val message: String) : Result()
+    }
+
+    @WorkerThread
+    suspend fun deleteAccount(userManager: UserManager): Result = withContext(Dispatchers.IO) {
```

Comment:
why worker thread annotation here?

Reply:
I think Deleting a user account can take a bit of time, and it can slow down or block the UI. So I used the worker thread annotation to handle it in the background.

Reply:
Worker thread annotation doesn't moves the job to background, it just alerts the developer to use the method on a background thread. first rule of coding : don't assume things